lib/rspec/core/bisect/example_minimizer.rb

```diff
@@ -136,7 +136,7 @@ def non_failing_example_ids
         end
 
         def get_expected_failures_for?(ids)
-          ids_to_run = ids + failed_example_ids
+          ids_to_run = all_example_ids & (ids + failed_example_ids)
```

Comment:
Nice trick!
If we wouldn’t have to support older Ruby versions, we could have used `intersection`.

What makes me wonder is that for later Ruby versions the method doc explicitly mentions that [the order is preserved](https://ruby-doc.org/core-2.7.0/Array.html#method-i-26), but not for [older ones](https://ruby-doc.org/core-1.8.7/Array.html#method-i-26).
I can’t tell just by looking at the C code if the stable ordering is true for older Ruby, too.

Do you think we have sufficient specs to cover the order?

Reply:
Thank you for your review!

> If we wouldn’t have to support older Ruby versions, we could have used intersection.

Yeah, and I also like `&`, it's very short :)

> Do you think we have sufficient specs to cover the order?

Yes, I think we have sufficient specs:
https://github.com/rspec/rspec-core/blob/v3.13.0/spec/rspec/core/bisect/example_minimizer_spec.rb#L13-L25
This spec fails if Ruby doesn't preserve the left array's order. And this repository tests Legacy Rubys.

Reply:
I found related issue:
https://bugs.ruby-lang.org/issues/7803
https://github.com/ruby/ruby/commit/933ff1d6d73eb43c1c57dc1ce8b25b172d7064bc

It seems to older Ruby has also preserved the order, and document reflected that implementation.